extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/conf/RadiusAuthenticationProtocol.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.apache.guacamole.auth.radius.conf;
+
+/**
+ * This enum represents supported RADIUS authentication protocols for
+ * the guacamole-auth-radius extension.
+ */
+public enum RadiusAuthenticationProtocol {
+    
+    // Password authentication protocol
+    PAP("pap"),
+    
+    // Challenge-Handshake AUthentication Protocol
+    CHAP("chap"),
+    
+    // Microsoft CHAP version 1
+    MSCHAPv1("mschapv1"),
+    
+    // Microsoft CHAP version 2
+    MSCHAPv2("mschapv2"),
+    
+    // Extensible authentication protocol with MD5 hashing.
+    EAP_MD5("eap-md5"),
+
+    // Extensible authentication protocol with TLS
+    EAP_TLS("eap-tls"),
+
+    // Extensible authentication protocol with Tunneled TLS
+    EAP_TTLS("eap-ttls");
+
+    // Store the string value used in the configuration file.
+    private final String strValue;
+    
+    /**
+     * Create a new RadiusAuthenticationProtocol object having the
+     * given string value.
+     * 
+     * @param strValue
+     *     The string value of the protocol.
+     */
+    public RadiusAuthenticationProtocol(String strValue) {
+        this.strValue = strValue;
+    }
+    
+    @Override
+    public String toString() {
+        return strValue;
+    }
+    
+    @Override
+    public static RadiusAuthenticationProtocol valueOf(String value) {
```

Comment:
`valueOf()` for enums is explicitly defined as returning the enum constant whose name exactly matches the given string:

https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#valueOf-java.lang.Class-java.lang.String-

I don't think we should override the default implementation of a function such that it no longer matches the contract of the original interface. This should probably be our own function.

Reply:
Again, not sure if you were looking at a specific commit instead of the latest round of code, but the latest has `getEnum()` defined (which I failed to document, so that needs fixing) - turns out that I don't think it's even valid to override `valueOf()` for `enum`s.